### PR TITLE
[4.0] [a11y] Table Options

### PR DESCRIPTION
**Pull request by brianteeman**
The table options drop down contains multiple select inputs
None of them are accessible as they don't have a label or have an invalid label

This issue was identified by the JAT 
> The ‘select’ elements do not have corresponding labels

This PR adds a label for each select with a class of sr-only (screenreader only)

Unfortunately there is no label attribute present in the xml so I had to add it. To avoid creating more work for translators I am reusing (where possible) the language string we are using for the default empty  value eg Select Position

### CODE REVIEW

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 20a908fe8b01835ac8a63155c708082399997b4b<br /><br />I have tested this item successfully but not with screenreader. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22008">issues.joomla.org/tracker/joomla-cms/22008</a>.</sub>

**Comment by brianteeman** [comment]
This is no longer a proof of concept.

@quy can you do your magic please

File: administrator/components/com_tags/forms/filter_tags.xml [hunk 1]
```diff
@@ -43,15 +47,17 @@
...
+			label="COM_TAGS_SELECT_TAGTYPE"
 			onchange="this.form.submit();"
 			>
 			<option value="">COM_TAGS_SELECT_TAGTYPE</option>
 		</field>
 	</fields>
 	<fields name="list">
 		<field
-            name="fullordering"
+        	name="fullordering"
```
Comment (Quy) [inline administrator/components/com_tags/forms/filter_tags.xml]: Indent

File: administrator/components/com_associations/forms/filter_associations.xml [hunk 2]
```diff
@@ -65,6 +69,7 @@
 		<field
 			name="level"
 			type="integer"
+			laabel="JOPTION_SELECT_MAX_LEVELS"
```
Comment (Quy) [inline administrator/components/com_associations/forms/filter_associations.xml]: spelling `laabel`

File: administrator/components/com_users/forms/filter_users.xml [hunk 3]
```diff
@@ -72,6 +77,7 @@
 		<field
 			name="limit"
 			type="limitbox"
+			label="JGLOBAL_LIST_LIMIT"
```
Comment (Quy) [inline administrator/components/com_users/forms/filter_users.xml:81]: Missing `label` attribute after line 55.

**Comment by ceus1984** [comment]
I have tested this item :red_circle: unsuccessfully on 71054f211ae86264a8d3138c3b0216904b714749<br /><br />I have tested it but cannot see any labels<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22008">issues.joomla.org/tracker/joomla-cms/22008</a>.</sub>

**Comment by ceus1984** [comment]
I have tested this item :red_circle: unsuccessfully on 71054f211ae86264a8d3138c3b0216904b714749<br /><br />I tried it but no labels appeared
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22008">issues.joomla.org/tracker/joomla-cms/22008</a>.</sub>

**Comment by brianteeman** [comment]
@ceus1984 
> This PR adds a label for each select with a class of sr-only (screenreader only)

So you will not be able to "see" the labels - you need to look at the source to "see" them or to use a screenreader

**Comment by SniperSister** [comment]
> So you will not be able to "see" the labels - you need to look at the source to "see" them or to use a screenreader

@brianteeman we did check the source code of the site, no labels in sight. Not sure if that's a side effect of using the patchtester to apply the change, but I indeed can confirm @ceus1984 observation cause I was looking over here shoulder

**Comment by brianteeman** [comment]
@SniperSister dont know what to tell you
image

**Comment by brianteeman** [comment]
conflicts fixed

**Comment by brianteeman** [comment]
~@infograf768 it is better if the labels are "Select" and not "- Select -"~

~So I can either create new strings for the label - disadvantage is that every extension will need to add the new strings to be accessible~
~Or I can change the strings to remove the " - " and then add the " - " in the select using css :before and :after disadvantage  if 3pd extensions dont update their strings they will see "-- Select --"~

I was mistaken screen readers dont announce the hyphen here

**Comment by brianteeman** [comment]
Closed - obviously no one else is interested in improving the accessibility in joomla 4

**Comment by brianteeman** [comment]
Trying again - maybe someone will look

**Comment by chmst** [comment]
I've tested again. With patchtester it was not successful. I don't know why. 
When I add the code manually into my branch, the test is successfull.

File: administrator/components/com_associations/forms/filter_associations.xml [hunk 4]
```diff
@@ -23,13 +24,13 @@
 			name="search"
 			type="text"
 			label="COM_ASSOCIATIONS_FILTER_SEARCH_LABEL"
-			description="COM_ASSOCIATIONS_FILTER_SEARCH_DESC"
```
Comment (Quy) [inline administrator/components/com_associations/forms/filter_associations.xml]: Remove language string?

Comment (brianteeman) [reply administrator/components/com_associations/forms/filter_associations.xml]: thanks - reverted - came from a merge conflict

File: administrator/components/com_workflow/forms/filter_stages.xml [hunk 5]
```diff
@@ -10,8 +10,7 @@
 		<field
 			name="published"
 			type="status"
-			label="COM_WORKFLOW_RADIO_CONDITION_LABEL"
-			description="COM_WORKFLOW_RADIO_CONDITION_LABEL_DESC"
```
Comment (Quy) [inline administrator/components/com_workflow/forms/filter_stages.xml:14]: There are other language strings that were removed in this PR. Delete them from the language file(s)?

Comment (brianteeman) [reply administrator/components/com_workflow/forms/filter_stages.xml:14]: Which ones? The only one I am aware of didn't exist

File: administrator/components/com_workflow/forms/filter_transitions.xml [hunk 6]
```diff
@@ -20,8 +19,7 @@
 		<field
 			name="from_stage"
 			type="sql"
-			label="COM_WORKFLOW_FROM_STAGE"
-			description="COM_WORKFLOW_FROM_STAGE_DESC"
```
Comment (Quy) [inline administrator/components/com_workflow/forms/filter_transitions.xml:24]: Delete?

File: administrator/components/com_workflow/forms/filter_transitions.xml [hunk 7]
```diff
@@ -31,8 +29,7 @@
 		<field
 			name="to_stage"
 			type="sql"
-			label="COM_WORKFLOW_TO_STAGE"
-			description="COM_WORKFLOW_TO_STAGE_DESC"
```
Comment (Quy) [inline administrator/components/com_workflow/forms/filter_transitions.xml:35]: Delete?

File: administrator/components/com_workflow/forms/filter_stages.xml (hunk 5 above)
Comment (brianteeman) [reply administrator/components/com_workflow/forms/filter_stages.xml:14]: These two strings are not in the language files

File: administrator/components/com_workflow/forms/filter_transitions.xml (hunk 6 above)
Comment (brianteeman) [reply administrator/components/com_workflow/forms/filter_transitions.xml:24]: yes this one can be

File: administrator/components/com_content/forms/filter_featured.xml [hunk 8]
```diff
@@ -12,6 +12,7 @@
 		<field
 			name="stage"
 			type="workflowstage"
+			label="COM_CONTENT_SELECT_STATE"
```
Comment (Quy) [inline administrator/components/com_content/forms/filter_featured.xml]: Should it be `JOPTION_SELECT_STAGE`?

Comment (brianteeman) [reply administrator/components/com_content/forms/filter_featured.xml]: Yes

File: administrator/components/com_menus/forms/filter_items.xml [hunk 9]
```diff
@@ -66,7 +71,6 @@
 			name="parent_id"
 			type="MenuItemByType"
 			label="COM_MENUS_FILTER_PARENT_MENU_ITEM_LABEL"
-			description="COM_MENUS_FILTER_PARENT_MENU_ITEM_DESC"
```
Comment (Quy) [inline administrator/components/com_menus/forms/filter_items.xml:69]: Remove? Last one I promise 😁

File: administrator/components/com_modules/forms/filter_modulesadmin.xml [hunk 10]
```diff
@@ -99,7 +97,7 @@
 		<field
 			name="limit"
 			type="limitbox"
-			label="COM_MODULES_LIST_LIMIT"
+			label="JGLOBAL_LIST_LIMIT"
 			description="JFIELD_PLG_SEARCH_SEARCHLIMIT_DESC"
```
Comment (Quy) [inline administrator/components/com_modules/forms/filter_modulesadmin.xml]: Remove `description`?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 6f3fb2e7101d07abb5edd909013eedbbda0cdca3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22008">issues.joomla.org/tracker/joomla-cms/22008</a>.</sub>

**Comment by ghost** [comment]
@chmst can you please retest?

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 6f3fb2e7101d07abb5edd909013eedbbda0cdca3<br /><br />I have tested this item successfully. The patchtester seems not to change the layout files, so this had to be done manually.
 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22008">issues.joomla.org/tracker/joomla-cms/22008</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

Thanks all for Tests.

**Comment by laoneo** [comment]
Thanks

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**